.github/workflows/reusable-porting-guide.yml

```
@@ -0,0 +1,92 @@
+---
+name: Create porting guide PR
+
+"on":
+  workflow_call:
+    inputs:
+      repository:
+        type: string
+        description: The repository to check out.
+        required: true
+      path:
+        type: string
+        description: The path for the repository.
+        required: true
+      ansible-version:
+        type: string
+        description: Ansible release version.
+        required: true
+      ansible-major-version:
+        type: string
+        description: Ansible major version.
+        required: true
+
+env:
+  GIT_BRANCH: porting-guide-${{ inputs.ansible-version }}
+  GITHUB_ACTOR: ${{ github.actor }}
+  ANSIBLE_VERSION: ${{ inputs.ansible-version }}
+  PORTING_GUIDE_RST: porting_guide_${{ inputs.ansible-major-version }}.rst
+  CI_COMMIT_MESSAGE: |-
+    Add the Ansible community ${{ inputs.ansible-version }} porting guide
+  PR_BODY_MESSAGE: |-
+    ##### SUMMARY
+
+    Add the Ansible ${{ inputs.ansible-major-version }} porting guide.
+
+    ##### ISSUE TYPE
+
+    - Docs Pull Request
+
+    ##### COMPONENT NAME
+
+    porting_guide_${{ inputs.ansible-major-version }}.rst
+
+jobs:
+  create-pr:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.path }}
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.repository }}
+        path: ${{ inputs.path }}
+
+    - name: Download the artifact with the RST file
+      uses: actions/download-artifact@v4
+      with:
+        name: ansible-porting-guide
+        path: ${{ inputs.path }}
+
+    - name: Copy the RST file to the docs repo
+      run: cp -v "${PORTING_GUIDE_RST}" docs/docsite/rst/porting_guides/
+
+    - name: Set up git
+      run: |
+        git switch --create ${GIT_BRANCH}
+        data=$(curl -s https://api.github.com/users/${GITHUB_ACTOR})
+        id=$(echo $data | jq .id)
+        name=$(echo $data | jq --raw-output '.name // .login')
```

This also lacks quoting:
```suggestion
        data="$(curl -s https://api.github.com/users/"${GITHUB_ACTOR}")"
        id="$(echo "${data}" | jq .id)"
        name="$(echo "${data}" | jq --raw-output '.name // .login')"
```

However, I'm sure this ID is present somewhere in the existing `${{ github }}` context, so there shouldn't be a need to make an API call over network in the first place.